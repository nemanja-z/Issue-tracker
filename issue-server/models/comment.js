module.exports =(sequelize, DataTypes) => {
    const Comment = sequelize.define("Comment", {
        comment:{
            type:DataTypes.STRING
          }
    });
    
    return Comment;
  };