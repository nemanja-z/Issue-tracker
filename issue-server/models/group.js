module.exports =(sequelize, DataTypes) => {
  const Group = sequelize.define('Group', {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    name: {
      type: DataTypes.STRING,
      unique: true
    },
  });
  Group.associate = (models) => {
    Group.belongsToMany(models.User, {
      through: 'member',
      foreignKey: {
        name: 'groupId',
        field: 'group_id'
      },
    });

    Group.belongsTo(models.User, {
      foreignKey: 'owner'
    });
  }
  return Group;
};