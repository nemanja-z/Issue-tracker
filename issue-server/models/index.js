import fs from 'fs';
import path from 'path';
import Sequelize from 'sequelize';
import configVar from '../config/config';
require('dotenv').config();
const basename = path.basename(__filename);
const env = process.env.NODE_ENV || 'development';
const config = configVar[env];
const db = {};
const cloudinary = require('cloudinary').v2;

let sequelize;
if (config.use_env_variable) {
  sequelize = new Sequelize(process.env[config.use_env_variable], config);
} else {
  sequelize = new Sequelize(config.database, config.username, config.password, config);
}
fs
  .readdirSync(__dirname)
  .filter(file => {
    return (file.indexOf('.') !== 0) && (file !== basename) && (file.slice(-3) === '.js');
  })
  .forEach(file => {
    const model = require(path.join(__dirname, file))(sequelize, Sequelize);
    db[model.name] = model;
  });

Object.keys(db).forEach(modelName => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

cloudinary.config({
  cloud_name: process.env.CLOUD_NAME,
  api_key: process.env.API_KEY,
  api_secret: process.env.API_SECRET
});


db.sequelize = sequelize;
db.Sequelize = Sequelize;
module.exports = {db, cloudinary};
