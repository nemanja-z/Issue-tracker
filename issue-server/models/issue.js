module.exports =(sequelize, DataTypes) => {
  const Issue = sequelize.define("Issue", {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    issue_type: {
      type: DataTypes.ENUM("Story", "Epic", "Bug", "Task"),
      allowNull: false
    },
    description: {
      type: DataTypes.STRING
    },
    summary: {
      type: DataTypes.STRING,
      allowNull:false
    },
    priority: {
      type: DataTypes.ENUM("Highest", "High", "Medium", "Low", "Lowest"),
      defaultValue: "Lowest"
    },
    resolution: {
      type: DataTypes.ENUM("Fixed","Won't do","Duplicate","Unresolved"),
      defaultValue: "Unresolved"
    }, 
    status:{
      type: DataTypes.ENUM("Reopened","Resolved","Closed","Active","Open"),
      defaultValue:"Open"
    },  
    reporter:{
      type: DataTypes.UUID,
        allowNull: false,
        references: {
          model: "User",
          key: "id"
        },
        onUpdate: 'CASCADE',
        onDelete: 'SET NULL',
    }
  });
   Issue.associate = models => {
    Issue.belongsTo(models.Project, {
      foreignKey: "project"
    });
    Issue.belongsToMany(models.User, {
      through: "Assignee"
    });
    Issue.hasMany(models.Comment, {
      foreignKey: 'issueId',
      constraints: false,
    });
  }; 
  return Issue;
};