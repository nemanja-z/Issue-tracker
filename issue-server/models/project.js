module.exports =(sequelize, DataTypes) => {
  const Project = sequelize.define("Project", {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false
    },
    url:{
      type:DataTypes.STRING
    }
  });
   Project.associate = models => {
    Project.hasMany(models.Issue,{
      foreignKey: "project"
    });
    Project.belongsToMany(models.User,{
      through:"Role"
    });
  }; 
  return Project;
};