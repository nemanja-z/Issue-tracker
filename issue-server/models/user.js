module.exports =(sequelize, DataTypes) => {
  const User = sequelize.define("User", {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    username: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
    },
    passwordHash: {
      type: DataTypes.STRING(64),
      is: /^[0-9a-f]{64}$/i,
      allowNull: false,
    }
  });
  User.associate = (models) => {
    User.belongsToMany(models.Project, {
      through: "Role"
    });
    User.belongsToMany(models.Issue, {
      through: "Assignee"
    });
}
  return User;
};