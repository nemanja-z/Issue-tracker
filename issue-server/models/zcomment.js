module.exports =(sequelize, DataTypes) => {
    const Comment = sequelize.define("Comment", {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true
      },
      comment:{
          type:DataTypes.STRING
        }
    });
    Comment.associate = models => {
      Comment.belongsTo(models.Issue, {
        foreignKey: "issueId",
        constraints: false
      });
      Comment.belongsTo(models.User, {
        as:"commenter"
      });
    }
    return Comment;
  };