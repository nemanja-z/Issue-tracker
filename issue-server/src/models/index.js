import { Sequelize } from 'sequelize';
import configVar from '../config/config';

const path = require('path');
const fs = require('fs');

require('dotenv').config();

const basename = path.basename(__filename);
const env = process.env.NODE_ENV || 'development';
const config = configVar[env];
const db = {};
const cloudinary = require('cloudinary').v2;


const sequelize = new Sequelize(process.env[config.use_env_variable]);

fs.readdirSync(__dirname)
  .filter((file) => (
    file.indexOf('.') !== 0 && file !== basename && file.slice(-3) === '.js'
  ))
  .forEach((file) => {
    const model = require(path.join(__dirname, file))(sequelize, Sequelize);
    db[model.name] = model;
  });

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

cloudinary.config({
  cloud_name: process.env.CLOUD_NAME,
  api_key: process.env.API_KEY,
  api_secret: process.env.API_SECRET,
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;
module.exports = { db, cloudinary };
