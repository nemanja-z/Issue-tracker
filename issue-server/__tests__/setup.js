const { ApolloServer, gql } = require('apollo-server');
import { createTestClient } from 'apollo-server-testing';
import dotenv from 'dotenv';
import path from 'path';
import { db as models } from '../models';
const { mergeTypeDefs } = require('@graphql-tools/merge');
const { mergeResolvers } = require('@graphql-tools/merge');
const { loadFilesSync } = require('@graphql-tools/load-files');
const bcrypt = require('bcrypt');

dotenv.config();

export const typeDefs = mergeTypeDefs(loadFilesSync(path.join(__dirname, '../schema')));
export const resolvers = mergeResolvers(loadFilesSync(path.join(__dirname, '../resolvers')));


  let user;
  const server = new ApolloServer({
    typeDefs,
    resolvers,
    context: () => {
      return {models,
      user}}
  });
  const {query, mutate} = createTestClient(server);
  

beforeAll(async()=>{
  await Promise.all(
    Object.keys(models).map(key => {
      if (['sequelize', 'Sequelize'].includes(key)) return null;      
      return models[key].destroy({ where: {}, force: true });
    })
  );
  user = await models.User.create({
    username: "Ljubivoje",
    email: "lginmwyffgkkpvgplk@niwghx.com",
    role: "Manager",
    profile:process.env.CLOUDINARY,
    passwordHash: await bcrypt.hash('Popajce', 10),
    isVerified:true
  })
  console.log({user})}
)
  describe('User', () => {
    /* const SIGN_UP = gql`
    mutation createUser($username:String!, $password:String!, $email:String!, $role:String!, $profile:Upload){
        createUser(username:$username, password:$password, email:$email, role:$role, profile:$profile){
            username
            email
            role
        }
    }`; */
  
      test('Login', async () => {
        const LOGIN = gql`
              mutation loginUser($username:String!, $password:String!){
              loginUser(username:$username, password:$password)
              }`;
        const loginUser = await mutate({
          mutation: LOGIN,
          variables: {
              username: 'Ljubivoje',
              password:'Popajce'
            
          }
        });
        expect(loginUser).toMatchSnapshot();
        // const { user } = result.data.deleteUser ?? {};
      });
  
    })

afterAll(async()=>{
      
    
      await models.sequelize.close();
  }
    )