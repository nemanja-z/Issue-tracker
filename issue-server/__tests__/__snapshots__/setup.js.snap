// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Login 1`] = `
Object {
  "data": Object {
    "loginUser": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6IkxqdWJpdm9qZSIsImlkIjoiODU1ODc3NWQtZjNlNC00YWMxLTkyZTgtODBlODIwMmU0MGIzIiwiaWF0IjoxNjA5MDc3MDM1fQ.qlSH1ELcwXpnVE5iVs-8XSOt8SkqIrhf3vV9NUnVVuE",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
